CMakeLists.txt

```diff
@@ -375,8 +387,8 @@ set(UNICORN_ARCH_COMMON
     qemu/accel/tcg/translator.c
 )
 
-if (UNICORN_HAS_X86)
-add_library(x86_64-softmmu
+if(UNICORN_HAS_X86)
+add_library(x86_64-softmmu STATIC
```

Comment:
Does `STATIC` here override `BUILD_SHARED_LIBS`?

Reply:
Yeah it does, but I don't see why you would event want to dynamically link the `*-softmmu` libraries. They are part of unicorn's private interface and all users can do something with is `unicorn.so`

In the previous CMake these were modeled by `OBJECT` libraries (likely an even better idea, since that means unicorn.lib will actually contain all the generated code).

Reply:
If you go to CMake 3.12 (released ~4 years ago) you can change `STATIC` to `OBJECT` and `target_link_libraries` will work seamlessly (in 3.1 you need to use `$<TARGET_OBJECTS:foo>` which is kind of nasty).

https://cmake.org/cmake/help/latest/command/add_library.html#object-libraries

Some people like to complain that their ancient distro doesn't have a recent CMake version, here is a decent post about it: https://alexreinking.com/blog/how-to-use-cmake-without-the-agonizing-pain-part-1.html

Reply:
> Yeah it does, but I don't see why you would ever want to dynamically link the `*-softmmu` libraries. They are part of unicorn's private interface and all users can do something with is `unicorn.so`

I indeed would like to keep `*-softmmu` as static libraries no matter what options users give. Sorry for the confusion.

> 
> In the previous CMake these were modeled by `OBJECT` libraries (likely an even better idea, since that means unicorn.lib will actually contain all the generated code).

This may solve https://github.com/unicorn-engine/unicorn/issues/1508#issuecomment-986094042 if I understand it correctly.

Reply:
Yes, switching to OBJECT libraries would solve this issue (since it's a cross platform way to merge everything into unicorn.lib)

Reply:
I think I'll leave this for now, getting linker errors about clashing symbols when attempting.